dspace-api/src/main/java/org/dspace/authorize/AuthorizeServiceImpl.java

```diff
@@ -787,4 +802,186 @@ public ResourcePolicy createOrModifyPolicy(ResourcePolicy policy, Context contex
         return resourcePolicyService.findExceptRpType(c, o, actionID, rpType);
     }
 
+    /**
+     * Checks that the context's current user is a community admin in the site by querying the solr database.
+     *
+     * @param context   context with the current user
+     * @return          true if the current user is a community admin in the site
+     *                  false when this is not the case, or an exception occurred
+     */
+    @Override
+    public boolean isCommunityAdminSolr(Context context) throws SQLException {
```

Comment:
it would be better to have a single method isCommunityAdmin that only check via SOLR. The code actually used for the database lookup should be moved in the indexer plugin that is mandatory to be used to have such implementation working properly. Having two different implementations is confusing and would lead to inefficiency as we could use the "old" one instead than the new one in most of the code base.

If you feel not comfortable with such change I suggest to create a code task for future improvement

Reply:
@Raf-atmire : I agree with @abollini here.  Would it be possible to simply *replace* the current `isCommunityAdmin()` methods with the contents of this new one?  I'd rather there be a single `isCommunityAdmin()` method which queries Solr, than have two different options.

Also, it's worth noting that the `isCommunityAdmin()` method which accesses the database seems to be primarily used by the REST API.  So, currently, parts of the REST API are using `isCommunityAdmin()` while other parts use `isCommunityAdminSolr()`.  This could result in odd behavior if the two are not identical.

If there is some reason why you chose this approach (of having two implementations), I'd be OK with moving this to a follow-up ticket.  But, I wanted to note that I don't like have two methods here doing the same thing in different ways.